scanner/src/main/kotlin/storages/FileBasedStorage.kt

```diff
@@ -20,15 +20,59 @@
 
 package com.here.ort.scanner.storages
 
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import com.here.ort.model.Identifier
 import com.here.ort.model.Package
+import com.here.ort.model.ScanResult
 import com.here.ort.model.ScanResultContainer
 import com.here.ort.model.ScannerDetails
+import com.here.ort.model.yamlMapper
 import com.here.ort.scanner.ScanResultsStorage
+import com.here.ort.utils.collectMessagesAsString
 import com.here.ort.utils.log
+import com.here.ort.utils.storage.FileStorage
+
+import java.io.IOException
+import java.io.PipedInputStream
+import java.io.PipedOutputStream
 
 const val SCAN_RESULTS_FILE_NAME = "scan-results.yml"
 
-abstract class FileBasedStorage : ScanResultsStorage() {
+/**
+ * A [ScanResultsStorage] using a [FileStorage] as backend. Scan results are serialized using [YAML][yamlMapper].
+ */
+class FileBasedStorage(
+    /**
+     * The [FileStorage] to use for storing scan results.
+     */
+    private val backend: FileStorage
```

Comment:
backend could be renamed to e.g. ```storage````

Reply:
I deliberately used a different name to not confuse both types of storages.